src/contexts/GustoApiProvider/GustoProviderWithUIAdapter.tsx

```diff
@@ -0,0 +1,39 @@
+import type React from 'react'
+import { ErrorBoundary } from 'react-error-boundary'
+import { I18nextProvider } from 'react-i18next'
+import { ReactSDKProvider } from '@gusto/embedded-api/ReactSDKProvider'
+import { ComponentsProvider } from '../ComponentAdapter/ComponentsProvider'
+import type { ComponentsContextType } from '../ComponentAdapter/ComponentsProvider'
+import { SDKI18next } from './SDKI18next'
+import { useGustoProvider, type GustoProviderProps } from './useGustoProvider'
+import { InternalError } from '@/components/Common'
+import { LocaleProvider } from '@/contexts/LocaleProvider'
+import { ThemeProvider } from '@/contexts/ThemeProvider'
+
+export interface GustoProviderWithUIAdapterProps extends GustoProviderProps {
+  children?: React.ReactNode
+}
+
+/**
+ * A provider that accepts UI component adapters through the components prop
+ */
+const GustoProviderWithUIAdapter: React.FC<GustoProviderWithUIAdapterProps> = props => {
+  const { children, ...providerProps } = props
+  const { config, lng, locale, currency, theme, components } = useGustoProvider(providerProps)
+
+  return (
+    <ComponentsProvider value={components as ComponentsContextType}>
```

Comment:
🤔 wonder why we end up needing to cast using `as` here? might be worth figuring out